aws/src/main/java/org/apache/iceberg/aws/metrics/SqsClientFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.aws.metrics;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.iceberg.aws.HttpClientProperties;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+
+/** Factory class for SQS client */
+public class SqsClientFactory implements Serializable {
```

Comment:
I think this should be `SqsMetricsReporterAwsClientFactory`, similar to the `S3FileIOAwsClientFactory`, each functional entity gets its own client factory, not each AWS client has its own client factory. What do you think?

https://github.com/apache/iceberg/blob/master/aws/src/main/java/org/apache/iceberg/aws/S3FileIOAwsClientFactories.java

Reply:
Also we want the same pattern as S3 to have a `Factories` so users can use their own factory if necessary

Reply:
I have introduced `SqsMetricsReporterAwsClientFactory` interface.